### PR TITLE
Jenkins pipeline setup

### PR DESCRIPTION
**Pull request by alexmaldonado22**
Creating a Jenkins pipeline to publish BetterCloud Scim2 library to private Nexus repo.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,73 @@
+pipeline {
+    agent any
+    environment {
+        NEXUS_URL   = 'https://nexus.onedatascan.io/nexus/content/repositories/thirdparty/'
+        NEXUS_CREDS = credentials('jenkins-maven')
```
Comment (jjprokash) [inline Jenkinsfile:5]: I am not sure that we should have these credentials in plain text.  Are there any security concerns here?  Can Jenkins pass this value in?

File: build.gradle [hunk 2]
```diff
@@ -74,62 +58,54 @@ subprojects {
...
+                url = nexusUrl
+                credentials {
+                    username = nexusUser
+                    password = nexusPassword
                 }
-
-                pom.project {
-                    name "SCIM 2.0 SDK for Spring"
+            }
+//            maven {
```
Comment (jjprokash) [inline build.gradle]: If this is not needed, should it be removed?

File: build.gradle [hunk 3]
```diff
@@ -74,62 +58,54 @@ subprojects {
...
-                                developer {
-                                    id "daniel-lynch"
-                                    name "Daniel Lynch"
-                                    email "dantlynch@gmail.com"
-                                }
-                        ]
+                        developer {
+                            id = 'daniel-lynch'
+                            name = 'Daniel Lynch'
+                            email = 'dantlynch@gmail.com'
```
Comment (jjprokash) [inline build.gradle:88]: Is this developer section necessary?

File: scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java [hunk 4]
```diff
@@ -276,7 +276,7 @@ public Path replace(final int index, final Filter valueFilter)
    * @param index The exclusive index of the endpoint path element.
    * @return A new path to a beginning portion of this path.
    * @throws IndexOutOfBoundsException if the index is out of range
-   *         (<tt>index &lt; 0 || index &gt; size()</tt>)
+   *         (<code>index &lt; 0 || index &gt; size()</code>)
```
Comment (jjprokash) [inline scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java:279]: Why are we changing the Javadocs?  We only want to make necessary modifications.

File: Jenkinsfile (hunk 1 above)
Comment (alexmaldonado22) [reply Jenkinsfile:5]: We're not hardcoding the credentials..we're just specifying for Jenkins to use 'jenkins-maven' account credentials.

File: build.gradle (hunk 2 above)
Comment (alexmaldonado22) [reply build.gradle]: It was in the original file so I left as is. Should I remove?

File: build.gradle (hunk 3 above)
Comment (alexmaldonado22) [reply build.gradle:88]: It was in the original file so I left as is. Should I remove?

File: scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java (hunk 4 above)
Comment (alexmaldonado22) [reply scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java:279]: <tt> tag is not supported in HTML5. See below documentation.

https://www.javatpoint.com/html-tt-tag

File: Jenkinsfile (hunk 1 above)
Comment (jjprokash) [reply Jenkinsfile:5]: Got it.  Thanks for the clarification

File: build.gradle (hunk 3 above)
Comment (jjprokash) [reply build.gradle:88]: Leave as was in the original

File: build.gradle (hunk 2 above)
Comment (jjprokash) [reply build.gradle]: I didn't see this in the original:

```
//            maven {
//                url = uri("${rootProject.projectDir}/scim2")
//            }
```

File: scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java (hunk 4 above)
Comment (jjprokash) [reply scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java:279]: You added `withJavadocJar()`.  If we remove that, do the compilation errors go away?

File: build.gradle (hunk 2 above)
Comment (alexmaldonado22) [reply build.gradle]: I removed. This was used to build locally.

File: scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java (hunk 4 above)
Comment (alexmaldonado22) [reply scim2-sdk-common/src/main/java/com/bettercloud/scim2/common/Path.java:279]: The original had to build with JavadocJar so did not want to remove.

**Comment by jjprokash** [comment]
All looks good.

**Pull request merged**